sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubScanConfigurationTest.java

```diff
@@ -24,4 +26,68 @@ void sechub_job_config_example1_JSON_can_be_desrialized_and_contains_expected_lo
         assertEquals("https://productfailure.demo.example.org/login", scanConfig.getWebScan().get().getLogin().get().getUrl().toExternalForm());
     }
 
+    @Test
+    void sechub_job_config_license_scan_JSON_can_be_deserialized_and_contains_expected_source_data_reference() {
+
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_license_scan_config_source_example.json"));
+
+        /* execute */
+        SecHubScanConfiguration scanConfig = SecHubScanConfiguration.createFromJSON(json);
+
+        /* test */
+        Set<String> usedDataConfigurations = scanConfig.getLicenseScan().get().getNamesOfUsedDataConfigurationObjects();
+        assertEquals(1, usedDataConfigurations.size());
+        assertEquals("code", usedDataConfigurations.iterator().next());
+    }
+
+    @Test
+    void sechub_job_config_license_scan_JSON_cannot_be_deserialized_used_singular_source() {
+
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_license_scan_wrong_source_config_example.json"));
+
+        /* execute + test */
+        Assertions.assertThrows(JSONConverterException.class, () -> {
+            SecHubScanConfiguration.createFromJSON(json);
+        });
+    }
+
+    @Test
+    void sechub_job_config_license_scan_JSON_non_existent_scan_type() {
+
+        /* prepare */
+        String json = TestFileReader.loadTextFile(new File("./src/test/resources/sechub_unknown_scan_type.json"));
+
+        /* execute */
+        SecHubScanConfiguration scanConfig = SecHubScanConfiguration.createFromJSON(json);
+
+        /* test */
+        assertFalse(scanConfig.getWebScan().isPresent());
+        assertFalse(scanConfig.getLicenseScan().isPresent());
+        assertFalse(scanConfig.getCodeScan().isPresent());
+        assertFalse(scanConfig.getInfraScan().isPresent());
+
+        assertEquals("1.0", scanConfig.getApiVersion());
+        assertEquals(1, scanConfig.getData().get().getSources().size());
+
+        SecHubSourceDataConfiguration sourceConfig = scanConfig.getData().get().getSources().iterator().next();
```

Comment:
The test is intended to test that no scan type is detected, but it can be read. You understood my intention. I also, wanted to test, that even if no proper scan type is defined the data section is still populated. The purpose of both is to test edge cases which the user might get wrong.